api/src/main/java/javax/enterprise/inject/spi/builder/InjectionPointConfigurator.java

```diff
@@ -0,0 +1,215 @@
+
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.spi.AnnotatedField;
+import javax.enterprise.inject.spi.AnnotatedParameter;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.enterprise.inject.spi.ProcessInjectionPoint;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.Set;
+
+/**
+ * This API is an helper to configure a new {@link InjectionPoint} instance.
+ * CDI container must provides an implementation of this interface.
+ *
+ * InjectionPointConfigurator is not reusable.
+ *
+ * This builder is not thread safe and shall not be used concurrently.
+ *
+ * @see ProcessInjectionPoint#configureInjectionPoint()
+ * @author Antoine Sabot-Durand
+ * @since 2.0
+ */
+public interface InjectionPointConfigurator {
+
+
+    /**
+     * Read the InjectionPoint information from the given {@link Field}.
+     * All relevant information is overwritten.
+     *
+     * @param field defining the InjectionPoint.
+     * @return self
+     */
+    InjectionPointConfigurator read(Field field);
+
+    /**
+     * Read the InjectionPoint information from the given {@link Method}.
+     * All relevant information is overwritten.
+     *
+     * @param method defining the InjectionPoint.
+     * @param param  the parameter's index in the Method defining the InjectionPoint
+     * @return self
+     */
+    InjectionPointConfigurator read(Method method, int param);
+
+    /**
+     * Read the InjectionPoint information from the given {@link Constructor}.
+     * All relevant information is overwritten.
+     *
+     * @param constructor defining the InjectionPoint.
+     * @param param       the parameter's index in the Constructor defining the InjectionPoint
+     * @return self
+     */
+    InjectionPointConfigurator read(Constructor<?> constructor, int param);
+
+    /**
+     * Read the InjectionPoint information from the given {@link AnnotatedField}.
+     * All relevant information is overwritten.
+     *
+     * @param field defining the InjectionPoint
+     * @return self
+     */
+    InjectionPointConfigurator read(AnnotatedField<?> field);
+
+    /**
+     * Read the InjectionPoint information from the given {@link AnnotatedParameter}.
+     * All relevant information is overwritten.
+     *
+     * @param param defining the InjectionPoint
+     * @return self
+     */
+    InjectionPointConfigurator read(AnnotatedParameter<?> param);
+
+    /**
+     * Read the InjectionPoint information from the given {@link InjectionPoint}.
+     * All relevant information is overwritten.
+     *
+     * @param delegate the InjectionPoint to get information from
+     * @return self
+     */
+    InjectionPointConfigurator read(InjectionPoint delegate);
+
+    /**
+     * Set the {@link Type} of the InjectionPoint to build.
```

Comment:
We should be more clear that this is **the required type** of injection point. I would also rename the param.